### PR TITLE
menu updates

### PR DESCRIPTION
**Pull request by karabutov**

### CODE REVIEW

File: bot/handlers.py [hunk 1]
```diff
@@ -330,22 +301,25 @@ def callback_query_handler(
...
+        user_id = update.effective_user.id
         if data == menu_items.SHOW_DEBTS:
-            token = context.user_data[States.CURRENT_EVENT_TOKEN]
-            user_id = update.effective_user.id
-            lenders_info, debtors_info = self._splitwise.get_final_transactions(token)
+            lenders_info, debtors_info = self._splitwise.get_final_transactions(event_token)
             if user_id in debtors_info:
-                update.callback_query.edit_message_text('Вы должны: \n' + str(debtors_info[user_id]))
+                update.callback_query.edit_message_text('Вы должны: \n' + str(debtors_info[user_id]) +
+                                                        f'\n\n {event.name}\nВыберете пункт меню:')
```
Comment (alexbuchkin) [inline bot/handlers.py]: Typo

File: bot/handlers.py [hunk 2]
```diff
@@ -330,22 +301,25 @@ def callback_query_handler(
...
             elif user_id in lenders_info:
-                update.callback_query.edit_message_text('Вам должны: \n' + str(lenders_info[user_id]))
+                update.callback_query.edit_message_text('Вам должны: \n' + str(lenders_info[user_id]) +
+                                                        f'\n\n {event.name}\nВыберете пункт меню:')
```
Comment (alexbuchkin) [inline bot/handlers.py]: Typo

File: bot/handlers.py [hunk 3]
```diff
@@ -330,22 +301,25 @@ def callback_query_handler(
...
             else:
-                update.callback_query.edit_message_text('Вы никому не должны и вам никто не должен')
-            update.effective_chat.send_message('Меню:', reply_markup=buttons.get_menu_keyboard())
+                update.callback_query.edit_message_text('Вы никому не должны и вам никто не должен!!!' +
+                                                        f'\n\n {event.name}\nВыберете пункт меню:')
```
Comment (alexbuchkin) [inline bot/handlers.py]: Typo

File: bot/handlers.py [hunk 4]
```diff
@@ -410,15 +349,14 @@ def fallbacks_handler(
...
-                          CommandHandler('add_expense', self.add_expense), ],
+            entry_points=[CallbackQueryHandler(self.callback_query_handler), ],
             states={
                 States.EXPENSE_NAME: [AddExpenseHandlers().get_conversation_handler()],
             },
             fallbacks=[MessageHandler(Filters.all, self.fallbacks_handler)],
             map_to_parent={
-                ConversationHandler.END: ConversationHandler.END
+                ConversationHandler.END: ConversationHandler.END,
+                States.EVENT_ACTIONS: States.EVENT_ACTIONS
```
Comment (alexbuchkin) [inline bot/handlers.py]: Please add a single comma to the end, like in previous line

File: bot/handlers.py [hunk 5]
```diff
@@ -445,7 +383,8 @@ def expense_name(
     ) -> States:
         expense_name = update.effective_message.text
         context.user_data[States.EXPENSE].name = expense_name
-        update.effective_chat.send_message('Введи потраченную сумму.')
+        update.effective_chat.send_message('Введи потраченную сумму или нажмите кнопку \'Отмена\'.',
```
Comment (alexbuchkin) [inline bot/handlers.py]: `Введи`, `нажмите` - please use same verb forms

File: bot/handlers.py [hunk 6]
```diff
@@ -515,6 +449,24 @@ def debt_sum(
...
+        update: Update,
+        context: CallbackContext,
+    ):
+        data = update.callback_query.data
+        if data == menu_items.CANCEL:
+            context.user_data.pop(States.EXPENSE, None)
+            context.user_data.pop(States.DEBT, None)
+            event_token = context.user_data[States.CURRENT_EVENT_TOKEN]
+            event = self._splitwise.get_event_info(event_token)
+            update.callback_query.edit_message_text(f'\n\n {event.name}\nВыберете пункт меню:')
```
Comment (alexbuchkin) [inline bot/handlers.py]: Typo

File: bot/handlers.py [hunk 7]
```diff
@@ -515,6 +449,24 @@ def debt_sum(
                                            reply_markup=buttons.get_user_buttons(users))
         return States.DEBTOR_NAME
 
+    def cancel_button_handler(
+        self,
+        update: Update,
+        context: CallbackContext,
+    ):
+        data = update.callback_query.data
+        if data == menu_items.CANCEL:
```
Comment (alexbuchkin) [inline bot/handlers.py:458]: What will happen if I send "cancel" as a message?

File: bot/handlers.py [hunk 8]
```diff
@@ -515,6 +449,24 @@ def debt_sum(
...
+            context.user_data.pop(States.EXPENSE, None)
+            context.user_data.pop(States.DEBT, None)
+            event_token = context.user_data[States.CURRENT_EVENT_TOKEN]
+            event = self._splitwise.get_event_info(event_token)
+            update.callback_query.edit_message_text(f'\n\n {event.name}\nВыберете пункт меню:')
+            update.callback_query.edit_message_reply_markup(reply_markup=buttons.get_event_commands_keyboard())
+            update.callback_query.answer('Закончили')
+            return States.EVENT_ACTIONS
+        return None
```
Comment (alexbuchkin) [inline bot/handlers.py:467]: Maybe `END` or smth instead of None?

File: bot/handlers.py [hunk 9]
```diff
@@ -531,16 +483,19 @@ def fallbacks_handler(
...
+                                     CallbackQueryHandler(self.cancel_button_handler)],
                 States.DEBTOR_NAME: [CallbackQueryHandler(self.debtor_name)],
                 States.DEBT_SUM: [MessageHandler(Filters.text, self.debt_sum)],
             },
             fallbacks=[CallbackQueryHandler(self.fallbacks_button_handler),
                        MessageHandler(Filters.all, self.fallbacks_handler)],
             map_to_parent={
-                ConversationHandler.END: ConversationHandler.END
+                ConversationHandler.END: ConversationHandler.END,
+                States.EVENT_ACTIONS: States.EVENT_ACTIONS
```
Comment (alexbuchkin) [inline bot/handlers.py]: Please add single comma to the end of the line

**Review by alexbuchkin — CHANGES_REQUESTED** [review]

File: bot/handlers.py (hunk 7 above)
Comment (karabutov) [reply bot/handlers.py:458]: fallback handler will process your message and nothing will happen

File: bot/handlers.py (hunk 8 above)
Comment (karabutov) [reply bot/handlers.py:467]: Conversation handler not change state when we return None

File: bot/handlers.py (hunk 9 above)
Comment (karabutov) [reply bot/handlers.py]: Done

File: bot/handlers.py (hunk 5 above)
Comment (karabutov) [reply bot/handlers.py]: Done

File: bot/handlers.py (hunk 4 above)
Comment (karabutov) [reply bot/handlers.py]: Done

File: bot/handlers.py [hunk 10]
```diff
@@ -70,7 +70,7 @@ def text_handler(
         update: Update,
         _: CallbackContext,
     ):
-        update.effective_chat.send_message('Введи команду или выберете пункт меню:')
+        update.effective_chat.send_message('Введи команду или выберите пункт меню:')
```
Comment (alexbuchkin) [inline bot/handlers.py]: Please use same verb forms

**Review by alexbuchkin — APPROVED** [review]

**Pull request merged**